src/mint.ts

```diff
@@ -246,7 +247,7 @@ export async function signMintTxns({
   txGroup: SignerTransaction[];
   initiatorSigner: InitiatorSigner;
 }): Promise<Uint8Array[]> {
-  const lsig = new LogicSigAccount(pool.program);
+  const {lsig} = pool.account;
```

Comment:
Same here. Isn't `pool.account` already `lsig` ?
```suggestion
  const lsig = pool.account;
```

```
export interface PoolInfo {
  account: LogicSigAccount;
  contractVersion: ContractVersionValue;
  validatorAppID: number;
  asset1ID: number;
  asset2ID: number;
  liquidityTokenID?: number;
  status: PoolStatus;
}
```